context.c

```diff
@@ -52,6 +54,7 @@ void ctx_free(struct Context **ptr)
     return;
 
   struct Context *ctx = *ptr;
+  struct Context *np = NULL, *tmp = NULL;
```

Comment:
I would move these near the point of use, i.e., just before the STAILQ macro call.

Reply:
no longer required, so dropped this.